lxd/instance/drivers/driver_qemu.go

```diff
@@ -7075,6 +7075,48 @@ func (d *qemu) checkFeatures(hostArch int, qemuPath string) (map[string]any, err
 		features["cpu_hotplug"] = struct{}{}
 	}
 
+	// Check AMD SEV features (only for x86 architecture)
+	if hostArch == osarch.ARCH_64BIT_INTEL_X86 {
+		cmdline, err := os.ReadFile("/proc/cmdline")
+		if err != nil {
+			return nil, err
+		}
+
+		parts := strings.Split(string(cmdline), " ")
+
+		// Check if SME is enabled in the kernel command line.
+		if shared.StringInSlice("mem_encrypt=on", parts) {
+			features["sme"] = struct{}{}
+		}
+
+		// Check if SEV/SEV-ES are enabled
+		sev, err := os.ReadFile("/sys/module/kvm_amd/parameters/sev")
+		if err != nil && !os.IsNotExist(err) {
+			return nil, err
+		} else if string(sev) == "Y\n" {
+			// Host supports SEV, check if QEMU supports it as well.
+			capabilities, err := monitor.SEVCapabilities()
+			if err != nil {
+				logger.Debug("Failed querying SEV capability during VM feature check", logger.Ctx{"err": err})
+			} else {
+				features["sev"] = struct{}{}
+				features["sev.cbitpos"] = capabilities.CBitPos
+				features["sev.reduced_phys_bits"] = capabilities.ReducedPhysBits
+
+				// If SEV is enabled on host and supported by QEMU,
+				// check if the SEV-ES extension is enabled.
+				sevES, err := os.ReadFile("/sys/module/kvm_amd/parameters/sev_es")
+				if err != nil {
+					logger.Debug("Failed querying SEV-ES capability during VM feature check", logger.Ctx{"err": err})
+				} else {
+					if string(sevES) == "Y" {
```

Comment:
Above you tested for `Y\n` when using os.ReadFile and here you're only testing for `Y`, is this a bug?

Perhaps for readability and safety we should pass both `/sys/module/kvm_amd/parameters/sev` and `/sys/module/kvm_amd/parameters/sev_es` through `strings.TrimSpace()` before comparison.